### PR TITLE
ddl: args v2 for truncate/exchange/alter partition

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: ref #53930 

Problem Summary:

### What changed and how does it work?
includes:
ActionTruncateTablePartition
ActionExchangeTablePartition
ActionAlterTablePartitionAttributes
ActionAlterTablePartitionPlacement
### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/ddl/executor.go [hunk 1]
```diff
@@ -2018,27 +2019,6 @@ func (e *executor) multiSchemaChange(ctx sessionctx.Context, ti ast.Ident, info
 				Table:    ref.RefTable.L,
 				Mode:     model.SharedInvolving,
 			})
-		case model.ActionAlterTablePartitionPlacement:
```
Comment (D3Hunter) [inline pkg/ddl/executor.go:2021]: it's not a valid multiple schema change action

**Review by joccau — APPROVED** [review]
LGTM

**Comment by D3Hunter** [comment]
br failed due to https://github.com/pingcap/tidb/issues/56308

**Pull request merged**